src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9728,65 +9727,13 @@ namespace SurfaceGeometry {
-                    ShowWarningError(state,
-                                     cCurrentModuleObject + "=\"" + windowShadingControl.Name + " has " + state.dataIPShortCut->cAlphaFieldNames(5) +
-                                         " \"" + ControlType + "\" but a schedule has not been specified.");
-                }
-            }
-
-            if (has_prefix(ControlType, "SCHEDULEAND")) {
-                ControlType = "ONIFHIGH" + ControlType.substr(11);
-                windowShadingControl.ShadingControlIsScheduled = true;
-                windowShadingControl.GlareControlIsActive = false;
-                ShowWarningError(state,
-                                 cCurrentModuleObject + "=\"" + windowShadingControl.Name + "\" is using obsolete " +
-                                     state.dataIPShortCut->cAlphaFieldNames(5) + "=\"" + state.dataIPShortCut->cAlphaArgs(5) + "\", changing to \"" +
-                                     ControlType + "\"");
-                // Error if schedule has not been specified
-                if (windowShadingControl.Schedule <= 0) {
-                    ErrorsFound = true;
-                    ShowWarningError(state,
-                                     cCurrentModuleObject + "=\"" + windowShadingControl.Name + " has " + state.dataIPShortCut->cAlphaFieldNames(5) +
-                                         " \"" + ControlType + "\" but a schedule has not been specified.");
-                }
-            }
-
-            if (has_prefix(ControlType, "GLAREOR")) {
-                ControlType = "ONIFHIGH" + ControlType.substr(7);
-                windowShadingControl.ShadingControlIsScheduled = false;
-                windowShadingControl.GlareControlIsActive = true;
-                ShowWarningError(state,
-                                 cCurrentModuleObject + "=\"" + windowShadingControl.Name + "\" is using obsolete " +
-                                     state.dataIPShortCut->cAlphaFieldNames(5) + "=\"" + state.dataIPShortCut->cAlphaArgs(5) + "\", changing to \"" +
-                                     ControlType + "\"");
-            }
-
-            if (ControlType == "GLARE") {
-                ControlType = "ONIFHIGHGLARE";
-                windowShadingControl.ShadingControlIsScheduled = false;
-                windowShadingControl.GlareControlIsActive = true;
-                ShowWarningError(state,
-                                 cCurrentModuleObject + "=\"" + windowShadingControl.Name + "\" is using obsolete " +
-                                     state.dataIPShortCut->cAlphaFieldNames(5) + "=\"" + state.dataIPShortCut->cAlphaArgs(5) + "\", changing to \"" +
-                                     ControlType + "\"");
-            }
+            WindowShadingControlType shadingControlType = static_cast<WindowShadingControlType>(
+                getEnumerationValue(WindowShadingControlTypeNamesUC, UtilityRoutines::MakeUPPERCase(state.dataIPShortCut->cAlphaArgs(5))));
```

Comment:
Changed `ControlType` from `std::string `to `WindowShadingControlType shadingControlType`. Used `getEnumerationValue`.

Reply:
Every time you change a string to an enumeration, you get five extra minutes of life.